Week2/homework/src/part1_app.js

```diff
@@ -0,0 +1,145 @@
+const prompt = require('prompt');
+const mysql = require('mysql');
+const util = require('util');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'new_world',
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+const input = util.promisify(prompt.get.bind(this));
+
+async function queryDatabase() {
+  let input_number = '';
+  prompt.start();
+  try {
+    console.log(`
+
+      Options:
+
+      1........What is the capital of country X ? (Accept X from user)
+      2........List all the languages spoken in the region Y (Accept Y from user)
+      3........Find the number of cities in which language Z is spoken (Accept Z from user)
+      4........Are there any countries that have A) Same official language B) Same region If yes, display those countries. If no, display TRUE or FALSE
+      5........List all the continents with the number of languages spoken in each continent
+
+      Make your choice =>>
+    `);
+    const options = await input(['option']);
+    input_number = options.option;
+
+    switch (input_number) {
+      case '1':
+        const firstOption = await input(['country']);
+        let country = firstOption.country;
+        const firstQuery = `SELECT e1.Name AS Country, e2.Name AS Capital FROM country AS e1 LEFT JOIN city AS e2 ON (e1.Capital = e2.ID) WHERE e1.name = ?;`;
+        connection.connect();
+        let resultsFirst = await execQuery(firstQuery, country);
+        console.log(`
+        `);
+        for (r of resultsFirst) {
+          console.log(JSON.parse(JSON.stringify(r)));
+        }
+        console.log(`
+        `);
+        connection.end();
+        break;
+
+      case '2':
+        const secondOption = await input(['region']);
+        let region = secondOption.region;
+        const secondQuery = `SELECT region, COUNT(language) AS Languages FROM (SELECT e1.region AS region, e2.Language AS Language FROM country AS e1 LEFT JOIN (SELECT * FROM countrylanguage WHERE isOfficial = 'T') AS e2 ON (e1.Code = e2.CountryCode)) AS l where region = ? GROUP BY region;`;
```

Comment:
Can you change this to list the countries instead of the count?

Reply:
How did I missed it :sweat_smile: . Probably I have missed the main point among writing the queries. I corrected it @saragerion . Thanks :pray: